### ISSUE TITLE
Shifts : Setting default time to local time of user

### ISSUE THREAD

**Issue by sreejakshetty** [fossasia/engelsystem#108]
![screenshot from 2016-06-15 17 41 28](https://cloud.githubusercontent.com/assets/14835922/16079718/a2403a28-3322-11e6-904a-b4489469cb31.png)
![screenshot from 2016-06-15 17 18 29](https://cloud.githubusercontent.com/assets/14835922/16079724/a720bee6-3322-11e6-9f69-6eaa72cb94a6.png)
As we can see the default time is set to berlin's local time . We need to change it to local time of the user.

**Comment by sreejakshetty** [comment]
![screenshot from 2016-06-16 12 25 37](https://cloud.githubusercontent.com/assets/14835922/16107949/8926ea3c-33bd-11e6-93cc-886503ffb5e4.png)

**initially  the time is set to server time .**

![screenshot from 2016-06-16 12 26 00](https://cloud.githubusercontent.com/assets/14835922/16107970/bb45a13e-33bd-11e6-9518-dac4c4bc9fb1.png)
once we press the **time image**  it is set to local system time.
![screenshot from 2016-06-16 12 26 14](https://cloud.githubusercontent.com/assets/14835922/16107980/d4daf54a-33bd-11e6-9cb4-894b261972dc.png)
This issue can be closed now .

**Comment by hongquan** [comment]
@kamishettysreeja25 When user visit the Shift list page the first time, the datetime filter will have a default value. How this value is determined?

**Comment by sreejakshetty** [comment]
> When user visit the Shift list page the first time, the datetime filter will have a default value. How this value >is determined?

@hongquan The value is server time . As the shifts are created by admin initially it shows the current time of server/admin . When we press now it is set to local system time of the user.
We can change the timezone of admin in config.php file . We can create a issue which can allow admin to change his timezone in admin settings page.

**Comment by hongquan** [comment]
Where did you look into to give the conclusion that that value is the current time of server time? See the pic below, it is not Berlin time.

![selection_100](https://cloud.githubusercontent.com/assets/314607/16141321/786ebb4e-3484-11e6-845b-130f95170cd3.png)

No matter wheter I set timezone in config.php to "Europe/Berlin" or "Asia/Ho_Chi_Minh", the default value in the filter is still the same (11:13 as in the screenshot).

**Comment by sreejakshetty** [comment]
@hongquan I am able to view the local time of berlin . 
![screenshot from 2016-06-17 10 48 34](https://cloud.githubusercontent.com/assets/14835922/16141422/2080c216-3479-11e6-9337-b3a9e164c0d6.png)
![screenshot from 2016-06-17 10 48 07](https://cloud.githubusercontent.com/assets/14835922/16141421/2080b208-3479-11e6-8a95-f6d8de69e865.png)

**Comment by hongquan** [comment]
You have to look into PHP code, to see how the value is determine, not by comparing with the clock like that. What if the match between the clock and what you see in the web is mere coincidence?

**Comment by sreejakshetty** [comment]
@hongquan 
![screenshot from 2016-06-20 21 43 58](https://cloud.githubusercontent.com/assets/14835922/16201699/37af3cec-3730-11e6-96e0-3bdfd9a01443.png)
![screenshot from 2016-06-20 21 44 30](https://cloud.githubusercontent.com/assets/14835922/16202234/70a1f7cc-3732-11e6-8637-4be8b323a91d.png)
The start time and end time are session variables.
Initially the start time is set to date("H:i") = > this gives the current time in the timezone.
Once we logged in we will see the local time of the defualt timezone . 
Since it is a session variable, when we change the default timezone to another timezone in config.php/we reload the page we will still get the same time displayed .

**Comment by hongquan** [comment]
> Initially the start time is set to date("H:i") = > this gives the current time in the timezone.

No, PHP `date` function doesn't care about timezone (http://php.net/manual/en/function.date.php). It simply return local time of server, i.e the time in the clock of the server, no matter which timezone the server is, or our _config.php_ is.

It means that if the server clock is 6:30, it does not matter that you set timezone in _config.php_ as Berlin or New Delhi , `date()` function still returns 6:30.

To sum up, we don't have to do anything in this issue (at least now).

**Comment by hongquan** [comment]
I think I made mistake. The timezone of PHP `date()` function is affected by `date_default_timezone_set`

**Comment by sreejakshetty** [comment]
@hongquan Yes , it is affected by `date()`function . date() gives the data or time of the default timezone set.
That is the reason when we change the timezone the time and data changes in shifts page .
